scripts/zkai_stwo_ai_route_layout_policy_gate.py

```
@@ -0,0 +1,999 @@
+def build_policy_metric_rows(
+    section_payload: dict[str, Any],
+    route_payload: dict[str, Any],
+) -> list[dict[str, Any]]:
+    route_by_id = route_rows_by_profile(route_payload)
+    section_profile_ids = [
+        require_str(row.get("profile_id"), f"section row {index} profile_id")
+        for index, row in enumerate(section_payload["profile_rows"])
+    ]
+    expected_route_ids = set(section_profile_ids) | set(ROUTE_MATRIX_NON_SECTION_PROFILE_IDS)
+    if set(route_by_id) != expected_route_ids:
+        raise StwoAiRouteLayoutPolicyGateError("route matrix profile set drift")
```

Ruff (0.15.15)

[warning] Avoid specifying long messages outside the exception class

(TRY003)